src/relay/backend/contrib/vsi_npu/codegen.cc

```diff
@@ -0,0 +1,425 @@
+        std::make_shared<CallbackExpr>(expr));
+    vxOpmap_tbl_[expr] = output_Opsetup;
+  }
+  VisitInferred(expr);
+  return vxOpmap_tbl_;
+}
+
+typedef void (*setup_operand_fun_ptr)(VxOpTable&, Expr&);
+
+template <typename T>
+void setup_operand(VxOpTable& vxOpmap_tbl_, Expr& expr) {
+  vxOpmap_tbl_[expr] = std::make_shared<T>(vxOpmap_tbl_[expr]->specs_[0],vxOpmap_tbl_[expr]->pCallbackexpr_);
+}
+
+#define DEFINE_NODE_ITEM(name, op) \
+  {name, setup_operand<op>}
+
+static std::map<std::string, setup_operand_fun_ptr> call_node_table = {
+  DEFINE_NODE_ITEM("nn.relu", Relu),
+  DEFINE_NODE_ITEM("nn.softmax", Softmax),
+  DEFINE_NODE_ITEM("nn.avg_pool2d", AvgPool),
+  DEFINE_NODE_ITEM("transpose", Transpose),
+  DEFINE_NODE_ITEM("qnn.add", QnnAdd),
+  DEFINE_NODE_ITEM("qnn.subtract", QnnSubtract),
+  DEFINE_NODE_ITEM("qnn.mul", QnnMul),
+  DEFINE_NODE_ITEM("maximum", Maximum),
+  DEFINE_NODE_ITEM("minimum", Minimum),
+  DEFINE_NODE_ITEM("nn.conv2d", Conv),
+  DEFINE_NODE_ITEM("qnn.quantize", Quantize),
+  DEFINE_NODE_ITEM("qnn.dequantize", Dequantize),
+  DEFINE_NODE_ITEM("reshape", Reshape),
+  DEFINE_NODE_ITEM("squeeze", Squeeze),
+  DEFINE_NODE_ITEM("argmax", ArgMax),
+  DEFINE_NODE_ITEM("argmin", ArgMin),
+  DEFINE_NODE_ITEM("image.resize2d", Resize),
+  DEFINE_NODE_ITEM("nn.max_pool2d", MaxPool2d),
+  DEFINE_NODE_ITEM("qnn.concatenate", VsiNpuConcat),
+  DEFINE_NODE_ITEM("add", Add),
+  DEFINE_NODE_ITEM("mean", Mean),
+  DEFINE_NODE_ITEM("sigmoid", Sigmoid),
+  DEFINE_NODE_ITEM("tanh", Tanh),
+  DEFINE_NODE_ITEM("nn.depth_to_space", DepthtoSpace),
+  DEFINE_NODE_ITEM("logical_and", LogicalAnd),
+  DEFINE_NODE_ITEM("logical_or", LogicalOr),
+  DEFINE_NODE_ITEM("nn.pad", Pad),
+  DEFINE_NODE_ITEM("nn.leaky_relu", LeakyRelu),
+  DEFINE_NODE_ITEM("qnn.requantize", QnnRequantize),
+};
+
+static std::map<std::string, setup_operand_fun_ptr> func_node_table = {
+  DEFINE_NODE_ITEM("vsi_npu.qnn_conv2d", VsiNpuQnnConv2d),
+  DEFINE_NODE_ITEM("vsi_npu.qnn_avgpool2d", VsiNpuQnnAvgPool),
+  DEFINE_NODE_ITEM("vsi_npu.qnn_softmax", VsiNpuQnnSoftmax),
+  DEFINE_NODE_ITEM("vsi_npu.qnn_sigmoid", VsiNpuQnnSigmoid),
+  DEFINE_NODE_ITEM("vsi_npu.qnn_clip", VsiNpuQnnClip),
+  DEFINE_NODE_ITEM("vsi_npu.qnn_dense", VsiNpuQnnDense),
+  DEFINE_NODE_ITEM("vsi_npu.qnn_mean", VsiNpuQnnMean),
+  DEFINE_NODE_ITEM("vsi_npu.qnn_leaky_relu", VsiNpuQnnLeakyRelu),
+  DEFINE_NODE_ITEM("vsi_npu.qnn_deconv", VsiNpuQnnDeconv),
+  DEFINE_NODE_ITEM("vsi_npu.qnn_tanh", VsiNpuQnnTanh),
+};
+
+void TensorMakerImpl::InferCall(const CallNode *cn) {
+  Call call_obj = GetRef<Call>(cn);
+  Expr expr = GetRef<Expr>(cn);
+  std::string name;
+  tvx::Quantization out_quant = tvx::Quantization();
+  if (const auto *fn = cn->op.as<FunctionNode>()) {
+    auto comp = fn->GetAttr<String>(attr::kComposite);
+    CHECK(comp.defined());
+    name = comp.value();
+    std::cout << "TensorMakerImpl::InferCall: " << name << std::endl;
```

Comment:
rm `std::cout` and replace with LOG calls if needed (see #9012 )

Reply:
will fix.